geode-redis/src/main/java/org/apache/geode/redis/internal/RedisCommandType.java

```diff
@@ -150,3016 +159,202 @@
-   */
-  ECHO {
-    private Executor executor;
-
-    @Override
-    public Executor getExecutor() {
-      if (executor == null) {
-        executor = new EchoExecutor();
-      }
-      return executor;
-    }
-
-    private final RedisDataType dataType = RedisDataType.NONE;
-
-    @Override
-    public RedisDataType getDataType() {
-      return this.dataType;
-    }
-  },
-
-  /**
-   * TIME
-   * <p>
-   * Return the current server time
-   */
-  TIME {
-    private Executor executor;
-
-    @Override
-    public Executor getExecutor() {
-      if (executor == null) {
-        executor = new TimeExecutor();
-      }
-      return executor;
-    }
-
-    private final RedisDataType dataType = RedisDataType.NONE;
-
-    @Override
-    public RedisDataType getDataType() {
-      return this.dataType;
-    }
-  },
-
-  /**
-   * PING
-   * <p>
-   * Ping the server
-   */
-  PING {
-    private Executor executor;
-
-    @Override
-    public Executor getExecutor() {
-      if (executor == null) {
-        executor = new PingExecutor();
-      }
-      return executor;
-    }
-
-    private final RedisDataType dataType = RedisDataType.NONE;
-
-    @Override
-    public RedisDataType getDataType() {
-      return this.dataType;
-    }
-  },
-
-  /**
-   * QUIT
-   * <p>
-   * Close the connection
-   */
-  QUIT {
-    private Executor executor;
-
-    @Override
-    public Executor getExecutor() {
-      if (executor == null) {
-        executor = new QuitExecutor();
-      }
-      return executor;
-    }
-
-    private final RedisDataType dataType = RedisDataType.NONE;
-
-    @Override
-    public RedisDataType getDataType() {
-      return this.dataType;
-    }
-  },
-
-  /**
-   * SHUTDOWN
-   * <p>
-   * Shut down the server
-   */
-  SHUTDOWN {
-    private Executor executor;
-
-    @Override
-    public Executor getExecutor() {
-      if (executor == null) {
-        executor = new ShutDownExecutor();
-      }
-      return executor;
-    }
-
-    private final RedisDataType dataType = RedisDataType.NONE;
-
-    @Override
-    public RedisDataType getDataType() {
-      return this.dataType;
-    }
-  },
-
-  // UNKNOWN
-  UNKNOWN {
-    private Executor executor;
-
-    @Override
-    public Executor getExecutor() {
-      if (executor == null) {
-        executor = new UnkownExecutor();
-      }
-      return executor;
-    }
-
-    private final RedisDataType dataType = RedisDataType.NONE;
-
-    @Override
-    public RedisDataType getDataType() {
-      return this.dataType;
-    }
-  };
-
-  /**
-   * Abstract method overridden by each value in enum to get the executor associated with that
-   * command type
-   *
-   * @return {@link Executor} for command type
-   */
-  public abstract Executor getExecutor();
-
-  public abstract RedisDataType getDataType();
-  /*
-   * private RedisCommandType (RedisDataType dataType) { this.dataType = dataType; }
-   */
+  private RedisCommandType(Executor executor, RedisDataType dataType) {
```

Comment:
I vaguely remembered that getDataType was unused. I double checked, and it is unused...do you want to get rid of that as well, or do you foresee some purpose for it?

Reply:
Yes. It is now gone. Thanks